Include total number of requests in test results

automatictester opened the pull request:
Fixes jmeter-maven-plugin/jmeter-maven-plugin#117

automatictester commented:
Hi, can this PR be merged and released in next version?

Ardesco commented:
We originally removed all of the reporting code from the meter-maven-plugin because it was having a massive performance hit.  Reporting on the tests was taking almost as long as running the tests.

What is left is minimal information to allow you to fail your build, we generally point people at other solutions if they want to make reporting better, there is also a jmeter maven analysis plugin which will perform much more in depth reporting and has been optimised to run quickly.

I agree that we should make it cleared what the numbers are because it can be confusing, but I have no desire to add in more reporting code.

automatictester commented:
I fully understand your concern about performance impact of reporting on overall test execution time.

I was looking at jmeter-maven-analysis-plugin, but this doesn't seem to be silver bullet here. Analysis plugin is a great thing for performance testing purposes. However, here I use JMeter for functional testing purposes. Due to wide range of supported interfaces this can be handy at times. Trying to find number of tests executed (and failed) in report created by that plugin is like looking for a needle in a haystack. Therefore I added it here - place which seems to be most natural for that kind of reporting.

I didn't notice performance deterioration after my change. In modified FailureScanner class I switched from Scanner to BufferedReader and addedd simple if statement to count not only failures, but also total number of results - this shouldn't have massive impact on test execution time. If you can send me (the.automatic.tester@gmail.com) some large JMeter result file, I can confirm my thoughts and provide you with exact numbers before this is merged. If there will be any negative performance impact, I can switch it back to Scanner and so some other tweaks - if needed.

Having this simple information about total number of tests would be of great benefit for all of use, who use JMeter coupled with your plugin for purposes other than performance testing.

Ardesco commented:
FailureScanner was originally using a BufferedReader.  It was removed because of performance problems.

automatictester commented:
Thanks for the clarification, I will switch my solution to use Scanner instead of BufferedReader. I will push and let you know when it's done.

automatictester commented:
I've changed the code to parse result file(s) with Scanner instead of BufferedReader. Everything should be fine now. Please let me know if you have any concerns about merging this.

automatictester commented:
Are you satisfied with the performance, since I switched to Scanner instead of BufferedReader? Can we merge that PR? If there is anything that still concerns you, let me know.

Ardesco commented:
I've not had a chance to try it yet, it's on my (very long) list of things to do.

automatictester commented:
Hey, did you have a chance to take a look at this? It would be great to have it released in upcoming version.

afranken commented:
I'll take  a look at this PR.

afranken commented:
I just opened a second pull requests with minor optimizations for the commits provided here: #132
